RHDEVDOCS-2836: Document on when expressions and final tasks of the finally section for Pipelines 1.5 (#34592)
openshift/openshift-docs: modules/op-about-finally_tasks.adoc

// This module is included in the following assembly:
//
// *openshift_pipelines/understanding-openshift-pipelines.adoc

[id="about-finally_tasks_{context}"]
= Finally tasks

> The `finally` tasks are the final set of tasks specified using the `finally` field in the pipeline YAML file. A `finally` task always executes the tasks within the pipeline, even if they fail. The `finally` tasks are executed in parallel after all the pipeline tasks are run, before the corresponding pipeline exits.

Author: Done. Changed to: A `finally` task always executes the tasks within the pipeline, irrespective of whether they are executed successfully.

> The `finally` tasks are the final set of tasks specified using the `finally` field in the pipeline YAML file. A `finally` task always executes the tasks within the pipeline, irrespective of whether they are executed successfully. The `finally` tasks are executed in parallel after all the pipeline tasks are run, before the corresponding pipeline exits.

Reviewer: > A `finally` task always executes the tasks within the pipeline, irrespective of whether they are executed successfully
what do `they` refer to?

Author: changing this entence to: A `finally` task always executes the tasks within the pipeline, irrespective of whether the pipeline runs are executed successfully.

You can configure a `finally` task to consume the results of any task within the same pipeline. This approach does not change the order in which this final task is run. It is executed in parallel with other final tasks after all the non-final tasks are executed.

> The following example shows a code snippet of the `clone-cleanup-workspace` pipeline, which clones the repository into a shared workspace and cleans up the workspace after executing all the final set of tasks as specified in the `finally` section of the pipeline YAML file.

Reviewer 2: ```suggestion
The following example shows a code snippet of the `clone-cleanup-workspace` pipeline. This snippet clones the repository into a shared workspace. After executing the tasks specified in the `finally` section of the pipeline YAML file, it cleans up the workspace.
```

Author: Done.

> The following example shows a code snippet of the `clone-cleanup-workspace` pipeline. This code clones the repository into a shared workspace and cleans up the workspace. After executing the tasks specified in the `finally` section of the pipeline YAML file, the `cleanup` task cleans up the workspace.

Author: @VeereshAradhya Thanks for the suggestion. But it is obvious from the YAML that the `cleanup` task is from the `finally` section; hence no need to specify it here in the sentence. Any particular reason why you will want to specify it here explicitly? What is the value add?

> The following example shows a code snippet of the `clone-cleanup-workspace` pipeline. This code clones the repository into a shared workspace and cleans up the workspace. After executing the tasks specified in the `finally` section of the pipeline YAML file, the `cleanup` task cleans up the workspace.

Author: WIll change it to: After executing the pipeline tasks, the `cleanup` task specified in the `finally` section of the pipeline YAML file cleans up the workspace.

          value: application
      workspaces:
        - name: output
          workspace: git-source
  finally:
    - name: cleanup <4>
      taskRef: <5>
        name: cleanup-workspace 
      workspaces: <6>
        - name: source
          workspace: git-source
    - name: check-git-commit
      params: <7>
        - name: commit
          value: $(tasks.clone-app-repo.results.commit)
      taskSpec: <8>
        params:
          - name: commit
        steps:
          - name: check-commit-initialized
            image: alpine
            script: |
              if [[ ! $(params.commit) ]]; then
                exit 1
              fi
----
<1> Unique name of the Pipeline.
<2> The shared workspace where the git repository is cloned.
<3> The task to clone the application repository to the shared workspace.
<4> The task to clean-up the shared workspace.
<5> A reference to the task that is to be executed in the TaskRun.
<6> A shared storage volume that a Task in a Pipeline needs at runtime to receive input or provide output.
<7> A list of parameters required for a task. If a parameter does not have an implicit default value, you must explicitly set its value.

> <8> Task definition that is to be embedded directly in the TaskRun.

Author: Will change it to: Task definition that is to be embedded directly in the Task.
Please confirm.

> <8> Task definition that is to be embedded directly in the TaskRun.

Reviewer 3: yes I like @VeereshAradhya  suggestion, and change corresponds to <8>
